server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -293,11 +293,13 @@ private RolloverResult rolloverDataStream(
             ? dataStream.getAutoShardingEvent()
             : switch (autoShardingResult.type()) {
                 case NO_CHANGE_REQUIRED -> {
-                    logger.info(
-                        "Rolling over data stream [{}] using existing auto-sharding recommendation [{}]",
-                        dataStreamName,
-                        dataStream.getAutoShardingEvent()
-                    );
+                    if (dataStream.getAutoShardingEvent() != null) {
+                        logger.info(
+                            "Rolling over data stream [{}] using existing auto-sharding recommendation [{}]",
+                            dataStreamName,
+                            dataStream.getAutoShardingEvent()
+                        );
+                    }
                     yield dataStream.getAutoShardingEvent();
                 }
```

Comment:
This `case` now looks pretty similar to the `COOLDOWN_PREVENTED_INCREASE, COOLDOWN_PREVENTED_DECREASE` one (except for the comment). Maybe we can combine them?